includes/admin/extensions/class-extension-product-data.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+namespace EDD\Admin\Extensions;
+
+class ProductData {
+
+	/**
+	 * The product name.
+	 *
+	 * @since 2.11.x
+	 * @var string
+	 */
+	public $title;
+
+	/**
+	 * The product slug.
+	 *
+	 * @since 2.11.x
+	 * @var string
+	 */
+	public $slug;
+
+	/**
+	 * The URL for the product featured image.
+	 *
+	 * @since 2.11.x
+	 * @var string
+	 */
+	public $image;
+
+	/**
+	 * The product description.
+	 *
+	 * @since 2.11.x
+	 * @var string
+	 */
+	public $description;
+
+	/**
+	 * The extension basename.
+	 *
+	 * @since 2.11.x
+	 * @var string
+	 */
+	public $basename;
+
+	/**
+	 * The settings tab where the extension settings will show.
+	 *
+	 * @since 2.11.x
+	 * @var string
+	 */
+	public $tab;
+
+	/**
+	 * The settings section for the extension.
+	 *
+	 * @since 2.11.x
+	 * @var string
+	 */
+	public $section;
+
+	/**
+	 * The product features.
+	 *
+	 * @since 2.11.x
+	 * @var array
+	 */
+	public $features = array();
+
+	/**
+	 * Take array and return object.
+	 *
+	 * @since 2.11.x
+	 * @param array $array
+	 * @return ProductData
```

Comment:
```suggestion
	 * @return ProductData
	 * @throws \InvalidArgumentException
```

(Assuming you accept my next suggestion.)